---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---
### New Imputation for EMBRAPA with DArT Marker only and DArT plus GBS Markers

Diversity Array Technology LTDA joint all the genotyping data from the four Genotyping Orders that EMBRAPA requested during these six or seven years in one huge file.

So let's check what we got at the DArT report for EMBRAPA DArT genotyping of 2022
```{r, eval = F}
library(genomicMateSelectR)
dir("data")
nskipvcf <- 2
nskipcounts <- 2
VCF2024 <- read.table(here::here("data", "Report_9669_VCF_Ref_Version6.txt"),
                      sep = "\t", header = T, skip = nskipvcf, comment.char = "",
                      check.names = F)
Counts2024 <- read.table(here::here("data",
                                    "SEQ_SNPs_counts_0_Target_extend_Ref.csv"),
                         sep = ",", header = T, skip = nskipcounts, check.names = F)
Counts2024[1:10,1:50]
Counts2024[Counts2024$PICSnp != 0,] %>% nrow()/2
table(Counts2024$Chrom_Cassava_v61)[-1] %>% sum()/2
VCF2024[1:10,1:30]
VCFNames <- colnames(VCF2024)[10:ncol(VCF2024)]

write.table(VCFNames, file = "data/VCFCloneNames.csv", sep = ",", quote = F, row.names = F)
genomicMateSelectR::convertDart2vcf(dartvcfInput = here::here("data", "Report_9669_VCF_Ref_Version6.txt"),
                                    dartcountsInput = here::here("data", "SEQ_SNPs_counts_0_Target_extend_Ref.csv"),
                                    nskipvcf = 2, nskipcounts = 2,
                                    outName = "output/DCas24_9669", ncores = 1)
```


```{r See How many markers are common to both VCF files, eval = F}


```

